src/Analyzers/CSharp/Tests/SimplifyLinqExpression/SimplifyLinqExpressionFixAllTests.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.Analyzers.UnitTests.SimplifyLinqExpression
+{
+    public partial class SimplifyLinqExpressionTests
+    {
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineDeclaration)]
+        public async Task FixAllInDocument()
+        {
+            await TestInRegularAndScriptAsync(
+@"
+using System;
+using System.Linq;
+using System.Collections.Generic;
+
+class C
+{
+    static void M()
+    {
+        IEnumerable<string> test = new List<string> { ""hello"", ""world"", ""!"" };
+        var test1 = {|FixAllInDocument:test.Where(x => x.Equals('!')).Any()|};
+        var test2 = test.Where(x => x.Equals('!')).SingleOrDefault();
+        var test3 = test.Where(x => x.Equals('!')).Last();
+        var test4 = test.Where(x => x.Equals('!')).Count();
+        var test5 = test.Where(x => x.Equals('!')).FirstOrDefault();
+    }
+}",
+@"
+using System;
+using System.Linq;
+using System.Collections.Generic;
+
+class C
+{
+    static void M()
+    {
+        IEnumerable<string> test = new List<string> { ""hello"", ""world"", ""!"" };
+        var test1 = test.Any(x => x.Equals('!'));
+        var test2 = test.SingleOrDefault(x => x.Equals('!'));
+        var test3 = test.Last(x => x.Equals('!'));
+        var test4 = test.Count(x => x.Equals('!'));
+        var test5 = test.FirstOrDefault(x => x.Equals('!'));
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineDeclaration)]
+        public async Task NestedInDocument()
+        {
+            await TestInRegularAndScriptAsync(
+@"
+using System;
+using System.Linq;
+using System.Collections.Generic;
+
+class C
+{
+    static void M()
+    {
+        var test = new List<string> { ""hello"", ""world"", ""!"" };
+        var test1 = {|FixAllInDocument:test.Where(x => x.Equals('!')).Any()|};
+        var test2 = test.Where(x => x.Equals('!')).SingleOrDefault();
+        var test3 = test.Where(x => x.Equals('!')).Last();
+        var test4 = test.Where(x => x.Equals('!')).Count();
```

Comment:
i'm not sure if vb requires a trailing select in a linq query.  for both c# and vb can you do ``(from x in y where ...).Any()` i want to make sure we do the right thing, or at least not crash.

Reply:
I believe those are exposed as `ITranslatedQueryOperation` but I'll add tests. You are correct that this is legal in VB
```vb
Dim test1 = (From x In test Where x.Equals("!")).Any()
```